### ISSUE TITLE
Imported packages and dependencies

### ISSUE THREAD

**Issue by MarkEdmondson1234** [samterfa/googlePackageMaker#8]
As discussed, for now just get it working with whatever packages you need.

But eventually, we will examine the packages imported to see if they can be minimised.  It makes it a lot easier in the future to manage less rather than more dependencies.

One surprising very heavy dependency is `stringr`, so that would be one of my first choices to replace with base R `grep()` etc. 

There is also a decision on whether to embrace tidyverse in general as that can be a heavy dependency, I tend to use it for more user friendly packages like `googleAnalyticsR` but not for developer packages.

**Comment by samterfa** [comment]
That's fair. I hadn't thought of the distinction of this being a developer package vs a more user-friendly one. I presume your recommendation carries forward to the derived packages as well? I'm all in on Tidyverse but haven't had to maintain a package over a long time period so I trust your recommendations.

**Comment by MarkEdmondson1234** [comment]
Yes dplyr 0.8 comes to mind where all the data processing needed to be redone.  If its a package that in turn will be dependended on then one needs to consider that those packages will need to import as well, and so on.  It also makes build times much longer.
